File: csharp_body_deref/Scripts/Map/MapScriptComponent.cs
// ==================== AoTTG2 cross-reference ====================
// Type: Map.MapScriptComponent
// Update status: unchanged since the previous dump
// Real method bodies (native C decompile): source/cpp/Map/MapScriptComponent.c
// Prior real C# source (older reference): Assets/Scripts/Map/MapScript/MapScriptComponent.cs
// Bodies below are empty IL2CPP metadata stubs; the logic lives in the native C.
// ===============================================================
using System.Collections.Generic;
using Il2CppDummyDll;
using Utility;

namespace Map;

[Token(Token = "0x2000278")]
public class MapScriptComponent : BaseCSVRowItem
{
	[Token(Token = "0x4000D7F")]
	[FieldOffset(Offset = "0x10")]
	[Order(1)]
	public string ComponentName;

	[Token(Token = "0x4000D80")]
	[FieldOffset(Offset = "0x18")]
	[Order(2)]
	public List<string> Parameters;

	[Token(Token = "0x6000F45")]
	[Address(RVA = "0x3FB5D70", Offset = "0x3FB5D70", VA = "0x3FB5D70")]
	public MapScriptComponent()
	{
	}
}

File: cpp/Map/MapScriptComponent.c
// Type: Map.MapScriptComponent
// Ghidra (System V/gcc) decompilation, IL2CPP structs + signatures applied.
// ---- AoTTG2 cross-reference ----
// Update status: unchanged since the previous dump
// C# structure: source/csharp/Scripts/Map/MapScriptComponent.cs
// Prior real C# source (older reference): Assets/Scripts/Map/MapScript/MapScriptComponent.cs
// --------------------------------

// Map.MapScriptComponent$$.ctor
// il2cpp: void Map_MapScriptComponent___ctor (Map_MapScriptComponent_o* __this, const MethodInfo* method);
// 0x40b5d70

void Map_MapScriptComponent___ctor(Map_MapScriptComponent_o *__this,MethodInfo *method)

{
  System_Collections_Generic_List_string__o *__this_00;
  
  if (g_data_057ac4e5 == '\0') {
    il2cpp_runtime_helper_023445d0(&MethodInfo_List_1_System_String);
    il2cpp_runtime_helper_023445d0(&TypeInfo_List_string);
    g_data_057ac4e5 = '\x01';
  }
  (__this->fields).ComponentName = (System_String_o *)**(undefined8 **)(g_data_057b9c00 + 0xb8);
  il2cpp_runtime_helper_022b4080(&__this->fields);
  __this_00 = (System_Collections_Generic_List_string__o *)il2cpp_runtime_helper_023052d0(TypeInfo_List_string);
  System_Collections_Generic_List_object____ctor
            ((System_Collections_Generic_List_object__o *)__this_00,MethodInfo_List_1_System_String);
  (__this->fields).Parameters = __this_00;
  il2cpp_runtime_helper_022b4080(&(__this->fields).Parameters);
  Utility_BaseCSVRowItem___ctor((Utility_BaseCSVRowItem_o *)__this,(MethodInfo *)0x0);
  return;
}


